azure-functions/acbs-function/helpers/date.format-year.test.js

```diff
@@ -0,0 +1,77 @@
+const { formatYear } = require('./date');
+
+describe('formatYear', () => {
+  const testData = [
+    {
+      description: 'should return the input as a string when 4 digit number',
+      mockValue: 2024,
+      expected: '2024',
+    },
+    {
+      description: 'should return the input when 4 digit number stored as a string',
+      mockValue: '2024',
+      expected: '2024',
+    },
+    {
+      description: 'should return the input when 2 digit number stored as a string',
+      mockValue: '24',
+      expected: '2024',
+    },
+    {
+      description: 'should return the input as a string when 2 digit number',
+      mockValue: 24,
+      expected: '2024',
+    },
+    {
+      description: 'should return `2999` when input is 999',
+      mockValue: 999,
+      expected: '2999',
+    },
+    {
+      description: 'should return `1000` when input is 1000',
+      mockValue: 1000,
+      expected: '1000',
+    },
+    {
+      description: 'should return `0` when input is -2000',
+      mockValue: -2000,
+      expected: '0',
+    },
+    {
+      description: 'should return `2000` when input is 0',
+      mockValue: 0,
+      expected: '2000',
+    },
+    {
+      description: 'should return `2000` when input is -0',
+      mockValue: -0,
+      expected: '2000',
+    },
+    {
+      description: 'should return `NaN` when input is ``',
```

Comment:
Is this the behaviour we're going with in other microservices with these helpers? Or is this just the current iteration of how we deal with invalid inputs?

Reply:
Essentially these tests are more establishing existing behaviour, so I can replicate it exactly - there's definitely some test cases that I'd change but I'm hestitant to do that without being able to run these functions in a dev environment

Reply:
Okay sure, no worries there in that case. In the other PRs do we have similar to this or have we updated to do anything like throw exceptions etc.?

Reply:
This one's more unique in that there's no existing tests to guide what behaviour - in portal-api I did introduce a function `getStartOfDateFromDayMonthYearStringsReplicatingMoment` which has some unexpected behaviour the tests relied on. I then raised DTFS2-7024 to intestigate and remove this at some point